File: Samples for Parallel Programming with the .NET Framework/[C#][VB][C++][F#]-Samples for Parallel Programming with the .NET Framework/C# and C++ and F# and VB/C#,C++,F#,VB/Raytracer/Raytracer_VisualBasic/Surfaces.vb
'--------------------------------------------------------------------------
' 
'  Copyright (c) Microsoft Corporation.  All rights reserved. 
' 
'  File: Surfaces.vb
'
'--------------------------------------------------------------------------

Public Class Surfaces
    ' Only works with X-Z plane.
    Public Shared ReadOnly CheckerBoard As Surface = New Surface(
        Function(pos) If(((Math.Floor(pos.Z) + Math.Floor(pos.X)) Mod 2 <> 0), New Color(1, 1, 1), New Color(0.02, 0.0, 0.14)),
        Function(pos) New Color(1, 1, 1),
        Function(pos) If(((Math.Floor(pos.Z) + Math.Floor(pos.X)) Mod 2 <> 0), 0.1, 0.5),
        150)

    Public Shared ReadOnly Shiny As Surface = New Surface(
        Function(pos) New Color(1, 1, 1),
        Function(pos) New Color(0.5, 0.5, 0.5),
        Function(pos) 0.7,
        250)

    Public Shared ReadOnly MatteShiny As Surface = New Surface(
        Function(pos) New Color(1, 1, 1),
        Function(pos) New Color(0.25, 0.25, 0.25),
        Function(pos) 0.7,
        250)
End Class

File: Samples for Parallel Programming with the .NET Framework/[C#][VB][C++][F#]-Samples for Parallel Programming with the .NET Framework/C# and C++ and F# and VB/C#,C++,F#,VB/Raytracer/Raytracer_CSharp/Surfaces.cs
//--------------------------------------------------------------------------
// 
//  Copyright (c) Microsoft Corporation.  All rights reserved. 
// 
//  File: Surfaces.cs
//
//--------------------------------------------------------------------------

using System;

namespace Microsoft.ParallelComputingPlatform.ParallelExtensions.Samples
{
    static class Surfaces
    {
        // Only works with X-Z plane.
        public static readonly Surface CheckerBoard =
            new Surface(
                delegate(Vector pos)
                {
                    return ((Math.Floor(pos.Z) + Math.Floor(pos.X)) % 2 != 0)
                 ? new Color(1, 1, 1)
                 : new Color(0.02, 0.0, 0.14);
                },
                delegate(Vector pos) { return new Color(1, 1, 1); },
                delegate(Vector pos)
                {
                    return ((Math.Floor(pos.Z) + Math.Floor(pos.X)) % 2 != 0)
                 ? .1
                 : .5;
                },
                150);



        public static readonly Surface Shiny =
            new Surface(
                delegate(Vector pos) { return new Color(1, 1, 1); },
                delegate(Vector pos) { return new Color(.5, .5, .5); },
                delegate(Vector pos) { return .7; },
                250);

        public static readonly Surface MatteShiny =
            new Surface(
                delegate(Vector pos) { return new Color(1, 1, 1); },
                delegate(Vector pos) { return new Color(.25, .25, .25); },
                delegate(Vector pos) { return .7; },
                250);

    }
}
